src/Compilers/CSharp/Portable/Binder/DecisionDagBuilder.cs

```diff
@@ -679,32 +681,23 @@ DagState uniqifyState(ImmutableArray<StateForCase> cases, ImmutableDictionary<Bo
                 var state = new DagState(cases, remainingValues);
                 if (uniqueState.TryGetValue(state, out DagState? existingState))
                 {
-                    var newRemainingValues = existingState.RemainingValues;
-                    bool changed = false;
+                    // We found an existing state that matches.  Update its set of possible remaining values
+                    // of each temp by taking the union of the sets on each incoming edge.
+                    var newRemainingValues = ImmutableDictionary<BoundDagTemp, IValueSet>.Empty;
```

Comment:
Could we use a `Builder` instead? I'd imagine that constructing immutable dictionaries incrementally with `SetItem` would create more allocations :-/

Reply:
It turns out that these dictionaries are typically very small (usually zero or one item). I don't know which would overall be faster, but a builder is certainly an option. I'm going to leave it this way in this PR though.

---
In reply to: [486774228](https://github.com/dotnet/roslyn/pull/47567#discussion_r486774228) [](ancestors = 486774228)

Reply:
Actually, I took your suggestion and used a builder.

---
In reply to: [487215991](https://github.com/dotnet/roslyn/pull/47567#discussion_r487215991) [](ancestors = 487215991,486774228)